model/item.cc

```diff
@@ -223,8 +223,12 @@ namespace minsky
   bool Item::visibleWithinGroup() const 
   {return !attachedToDefiningVar();}
 
+  bool Item::inputDefined() {
+    if(m_ports.size()==0) return false;
+    auto vv=m_ports[0]->getVariableValue();
```

Comment:
ports[0] is the output, you will want to use ports[1] and or ports[2]

Reply:
Given some ports could be output, we should probably call this method inputIsScalar, then loop over all ports, and if the port is an input port (an attribute of the port), the port is either unconnected, or is scalar (rank 0).

Reply:
That way, the method needn't be virtual, as it is fully general.